### PR TITLE
신규 포스팅 작성 : OpenAI의 새로운 오픈소스 모델 패밀리, GPT OSS를 환영합니다!

### PR DESCRIPTION
**Pull request by choincnp**
개인사로 작업이 늦어져서 죄송합니다, 조금 더 문장을 native하게 번역하려고 노력했습니다!

### CODE REVIEW

File: _posts/2025-11-3-Welcome-GPT-OSS.md [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+    return_tensors="pt",
+    return_dict=True,
+).to(model.device)
+
+generated = model.generate(**inputs, max_new_tokens=100)
+print(tokenizer.decode(generated[0][inputs["input_ids"].shape[-1]:]))
+```
+
+### Flash Attention 3
+이 모델들은 attention sinks 기법을 사용하는데, 이는vLLM 팀이 Flash Attention 3와 호환되도록 만든 기법입니다. 우리는 그들의 최적화된 커널을 [kernels-community/vllm-flash-attn3](https://huggingface.co/kernels-community/vllm-flash-attn3)에 패키징하고 통합했습니다. 작성 시점 기준으로 이 초고속 커널은 PyTorch 2.7과 2.8에서 Hopper 카드로 테스트되었습니다. Hopper 카드(예: H100 또는 H200)에서 모델을 실행하는 경우, `pip install --upgrade kernels`를 실행하고 코드에 다음 줄을 추가해야 합니다:
```
Comment (hyeonseo2) [inline _posts/2025-11-3-Welcome-GPT-OSS.md]:
```suggestion
이 모델들은 attention sinks 기법을 사용하는데, 이는 vLLM 팀이 Flash Attention 3와 호환되도록 만든 기법입니다. 우리는 그들의 최적화된 커널을 [kernels-community/vllm-flash-attn3](https://huggingface.co/kernels-community/vllm-flash-attn3)에 패키징하고 통합했습니다. 작성 시점 기준으로 이 초고속 커널은 PyTorch 2.7과 2.8에서 Hopper 카드로 테스트되었습니다. Hopper 카드(예: H100 또는 H200)에서 모델을 실행하는 경우, `pip install --upgrade kernels`를 실행하고 코드에 다음 줄을 추가해야 합니다:
```

File: _posts/2025-11-3-Welcome-GPT-OSS.md [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+print(response)
+```
+
+# 로컬 실행
+
+## Transformers 사용하기
+최신 transformers 릴리스(v4.55.1 이상)와 함께 `accelerate`, `kernels`를 설치해야 합니다. 또한 triton 3.4 이상을 설치하는 것을 권장하는데, 이를 통해 CUDA 하드웨어에서 mxfp4 양자화를 지원할 수 있습니다.
+
+```bash
+pip install --upgrade accelerate transformers kernels
```
Comment (hyeonseo2) [inline _posts/2025-11-3-Welcome-GPT-OSS.md]:
```suggestion
pip install --upgrade accelerate transformers kernels "triton>=3.4"
```

원문에서 "triton>=3.4" 설치가 포함되어 있어서 추가했습니다!

File: _posts/2025-11-3-Welcome-GPT-OSS.md [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+---
+* TOC
+{:toc}
+<!--toc-->
+
+_이 글은 Hugging Face 블로그의 [Welcome GPT OSS, the new open-source model family from OpenAI!](https://huggingface.co/blog/welcome-openai-gpt-oss)를 한국어로 번역한 글입니다._
+
+---
+# 들어가며
+GPT OSS는 OpenAI가 공개한 대망의 오픈 웨이트 모델로, 강력한 추론 능력과 에이전트 태스크, 그리고 다양한 개발자의 사용 사례를 위해 설계되었습니다.
```
Comment (hyeonseo2) [inline _posts/2025-11-3-Welcome-GPT-OSS.md]:
```suggestion
GPT OSS는 OpenAI가 공개한 대망의 오픈 가중치(open-weights) 모델로, 강력한 추론 능력과 에이전트 작업, 그리고 다양한 개발자의 사용 사례를 위해 설계되었습니다. 
```

영문 표기와 병기하고, `task`를 이후에 번역된 `작업`으로 맞췄습니다! 참고 부탁드립니다!

**Review by hyeonseo2 — COMMENTED** [review]
좋은 번역 감사합니다 👍  아래 몇 가지 리뷰를 남겼습니다!

File: _posts/2025-11-3-Welcome-GPT-OSS.md [hunk 4]
```diff
@@ -0,0 +1,483 @@
...
+
+이 모델은 두 가지 버전으로 구성되었습니다: 
+- 117B 파라미터의 대형 모델([gpt-oss-120b](https://hf.co/openai/gpt-oss-120b))
+- 21B 파라미터의 소형 모델([gpt-oss-20b](https://hf.co/openai/gpt-oss-20b))
+
+두 모델 모두 혼합 전문가(Mixture-of-Experts, MoEs) 구조이며 4비트 양자화 방식(MXFP4)을 사용했기 때문에, 활성화되는 파라미터가 적어 빠른 추론이 가능하면서도 리소스 사용량은 낮게 유지됩니다. 대형 모델(gpt-oss-120b)은 H100 GPU 한 장에 올릴 수 있고, 소형 모델(gpt-oss-20b)은 16GB 메모리 내에서 구동되어 소비자용 하드웨어와 온디바이스 애플리케이션에 알맞습니다.
+
+커뮤니티에 더 나은 영향력을 제공하기 위해, 이 모델들은 최소한의 사용 정책과 함께 Apache 2.0 라이선스로 공개되었습니다:
+
+> 우리는 도구가 안전하고 책임감 있게, 그리고 민주적으로 사용되기를 바라며, 동시에 사용 방식에 대한 여러분의 통제권을 최대화하고자 합니다. gpt-oss를 사용함으로써 모든 관련 법규를 준수하는 데 동의하게 됩니다.
```
Comment (eehyo) [inline _posts/2025-11-3-Welcome-GPT-OSS.md:26]:
```suggestion
> 우리는 우리의 도구가 안전하고 책임감 있으며 민주적으로 사용되기를 바라며, 동시에 여러분이 이를 어떻게 사용할지에 대한 최대한의 통제권을 갖기를 원합니다. gpt-oss를 사용하는 경우, 관련 법규를 모두 준수해야 합니다.
```
좀 더 자연스럽게 수정해보았습니다!

File: _posts/2025-11-3-Welcome-GPT-OSS.md [hunk 5]
```diff
@@ -0,0 +1,483 @@
...
+- 각 어텐션 레이어는 128K 컨텍스트에 RoPE 사용
+- 전체 컨텍스트와 128 토큰 슬라이딩 윈도우를 가지는 어텐션 레이어 교차 배치
+- 어텐션 레이어는 헤드당 학습된 어텐션 싱크 사용 (softmax 분모에 추가적인 가산 값 적용)
+- GPT-4o 및 기타 OpenAI API 모델과 동일한 토크나이저 사용
+- Responses API와의 호환성을 위해 일부 새로운 토큰 추가
+
+![](https://i.imgur.com/6Kc55rS.png)
+`o3`, `o4-mini`와 비교한 OpenAI GPT OSS model들의 벤치마크 결과([OpenAI](https://openai.com/open-models/)제공).
+
+# 추론 프로바이더(Inference Providers)를 통한 API 접근
```
Comment (eehyo) [inline _posts/2025-11-3-Welcome-GPT-OSS.md]:
```suggestion
# Inference Providers를 통한 API 액세스
```
Inference Providers가 서비스 이름처럼 고유명사에 가깝다는 느낌이 들어서, 영어 그대로 사용하는 방향으로 수정해보았습니다. 편하신 방식으로 선택해 주세요! 아래 내용도 모두 동일하게 반영해두었습니다.

File: _posts/2025-11-3-Welcome-GPT-OSS.md [hunk 6]
```diff
@@ -0,0 +1,483 @@
...
+OpenAI GPT OSS 모델은 Hugging Face의 [Inference Providers](https://huggingface.co/docs/inference-providers/en/index) 서비스를 통해 접근할 수 있어, 동일한 자바스크립트나 파이썬 코드로 지원되는 모든 프로바이더에 요청을 보낼 수 있습니다. 이것은 OpenAI 공식 데모인 [gpt-oss.com](http://gpt-oss.com)을 구동하는 것과 동일한 인프라이며, 여러분의 프로젝트에서도 활용할 수 있습니다.
+
+다음은 파이썬과 초고속 Cerebras 프로바이더를 사용하는 예제입니다. 더 많은 정보와 추가 예제는 [모델 카드의 inference providers 섹션](https://huggingface.co/openai/gpt-oss-120b?inference_api=true&inference_provider=auto&language=python&client=openai)과 [gpt-oss 모델들을 위한 가이드](https://huggingface.co/docs/inference-providers/guides/gpt-oss)를 확인하세요.
```
Comment (eehyo) [inline _posts/2025-11-3-Welcome-GPT-OSS.md]:
```suggestion
다음은 Python과 매우 빠른 Cerebras provider를 사용하는 예제입니다. 더 많은 정보와 추가 예제는 [모델 카드의 inference providers 섹션](https://huggingface.co/openai/gpt-oss-120b?inference_api=true&inference_provider=auto&language=python&client=openai)과 [gpt-oss 모델들을 위한 가이드](https://huggingface.co/docs/inference-providers/guides/gpt-oss)를 확인하세요.
```

File: _posts/2025-11-3-Welcome-GPT-OSS.md [hunk 7]
```diff
@@ -0,0 +1,483 @@
...
+
+tokenizer.apply_chat_template(
+    chat,
+    model_identity="You are OpenAI GPT OSS.",
+    reasoning_effort="high"  # 기본값은 "medium"이지만, "high"와 "low"도 허용
+)
+```
+
+## Transformers를 사용한 도구 사용
+GPT OSS는 두 가지 종류의 도구를 지원합니다: "내장" 도구인 browser와 python, 그리고 사용자가 제공하는 커스텀 도구입니다. 내장 도구를 활성화하려면 아래처럼 채팅 템플릿의 `builtin_tools` 인자에 이름 목록을 리스트로 전달하세요. 커스텀 도구를 전달하려면, `tools` 인자를 사용하여 JSON 스키마 형식으로 전달하거나, 타입 힌트와 docstring을 포함한 파이썬 함수로 전달할 수 있습니다. 자세한 내용은 [채팅 템플릿 도구 문서](https://huggingface.co/docs/transformers/en/chat_extras)를 참조하거나, 아래 예제를 수정하면 됩니다:
```
Comment (eehyo) [inline _posts/2025-11-3-Welcome-GPT-OSS.md]:
```suggestion
GPT OSS는 두 가지 종류의 도구를 지원합니다: "내장" 도구인 browser와 python, 그리고 사용자가 제공하는 커스텀 도구입니다. 내장 도구를 활성화하려면 아래처럼 채팅 템플릿의 `builtin_tools` 인자에 이름 목록을 리스트로 전달하세요. 커스텀 도구를 전달하려면, `tools` 인자를 사용하여 JSON 스키마 형식으로 전달하거나, 타입 힌트와 docstring을 포함한 Python 함수로 전달할 수 있습니다. 자세한 내용은 [채팅 템플릿 도구 문서](https://huggingface.co/docs/transformers/en/chat_extras)를 참조하거나, 아래 예제를 수정하면 됩니다:
```

File: _posts/2025-11-3-Welcome-GPT-OSS.md [hunk 8]
```diff
@@ -0,0 +1,483 @@
...
+    add_generation_prompt=True,
+    return_tensors="pt",
+    return_dict=True,
+).to(model.device)
+
+generated = model.generate(**inputs, max_new_tokens=100)
+print(tokenizer.decode(generated[0][inputs["input_ids"].shape[-1]:]))
+```
+
+이 코드는 [이전 블로그 포스트](https://huggingface.co/blog/hello-hf-kernels)에서 설명한 대로 `kernels-community`에서 최적화되고 프리컴파일된 커널 코드를 다운로드합니다. transformers 팀이 코드를 빌드하고 패키징하고 테스트했으니 안심하고 사용하셔도 됩니다.
```
Comment (eehyo) [inline _posts/2025-11-3-Welcome-GPT-OSS.md:173]:
```suggestion
이 코드는 [이전 블로그 포스트](https://huggingface.co/blog/hello-hf-kernels)에서 설명한 대로 `kernels-community`에서 최적화된 사전 컴파일 커널 코드를 다운로드합니다. transformers 팀이 코드를 빌드하고, 패키징하고, 테스트까지 완료했기 때문에 안심하고 사용하셔도 됩니다.
```

**Review by eehyo — CHANGES_REQUESTED** [review]
좋은 번역 감사합니다! 주요 용어들을 통일하여 수정해두었습니다 :) 마지막에 Acknowledgements 부분이 생략되어 있는 것으로 보이는데 혹시 의도하신 건지 확인 부탁드립니다!

File: _posts/2025-11-3-Welcome-GPT-OSS.md [hunk 9]
```diff
@@ -0,0 +1,483 @@
...
+
+# 파인튜닝
+GPT OSS 모델은 `trl`과 완전히 통합되어 있습니다. Hugging Face에서는 `SFTTrainer`를 사용한 몇 가지 파인튜닝 예제를 개발했습니다:
+- [OpenAI cookbook의 LoRA 예제](https://cookbook.openai.com/articles/gpt-oss/fine-tune-transfomers) - 모델을 여러 언어로 추론하도록 파인튜닝하는 방법을 보여줍니다.
+- [기본 파인튜닝 스크립트](https://github.com/huggingface/gpt-oss-recipes/blob/main/sft.py) - 필요에 맞게 수정할 수 있습니다.
+
+# Hugging Face 파트너에 배포하기
+
+## Azure
+Hugging Face는 Azure와 협력하여 Azure AI Model Catalog를 통해 텍스트, 비전, 음성, 멀티모달 작업을 아우르는 가장 인기 있는 오픈소스 모델들을 고객 환경에 직접 제공합니다. 이를 통해 Azure의 엔터프라이즈급 인프라, 자동 확장, 모니터링을 활용하여 관리형 온라인 엔드포인트에 안전하게 배포할 수 있습니다.
```
Comment (ahnjj) [inline _posts/2025-11-3-Welcome-GPT-OSS.md]:
```suggestion
Hugging Face는 Azure와 협력하여 Azure AI Model Catalog를 이용해, 텍스트, 비전, 음성, 멀티모달 작업을 아우르는 
인기 있는 오픈소스 모델들을 고객 환경에 직접 제공합니다. 이를 통해 Azure의 엔터프라이즈급 인프라, 자동 확장, 모니터링을 활용하여 관리형 온라인 엔드포인트에 안전하게 배포할 수 있습니다.
```
문장이 길어서 둘로 나누어보았어요~! 어떠실까요?

File: _posts/2025-11-3-Welcome-GPT-OSS.md [hunk 10]
```diff
@@ -0,0 +1,483 @@
...
+## Dell
+Dell Enterprise Hub는 Dell 플랫폼을 사용하여 최신 오픈 AI 모델을 온프레미스에서 훈련하고 배포하는 과정을 간소화하는 보안 온라인 포털입니다. Dell과 협력하여 개발된 이 허브는 최적화된 컨테이너, Dell 하드웨어에 대한 네이티브 지원, 엔터프라이즈급 보안 기능을 제공합니다.
+
+GPT OSS 모델은 이제 [Dell Enterprise Hub](https://dell.huggingface.co/)에서 사용 가능하며, Dell 플랫폼을 사용해서 온프레미스에 바로 배포할 수 있습니다.
+
+![](https://i.imgur.com/Pcva15s.png)
+
+
+# 모델 평가하기
+GPT OSS 모델은 추론 모델입니다. 따라서 평가 시 매우 큰 생성 크기(최대 새 토큰 수)가 필요합니다. 모델의 생성 결과에는 먼저 추론 과정이 포함되고, 그 다음에 실제 답변이 나오기 때문입니다. 생성 크기가 너무 작으면 추론 중간에 예측이 중단될 위험이 있으며, 이는 위음성을 발생시킬 수 있습니다. 메트릭을 계산하기 전에 모델 답변에서 추론 과정을 제거해야 하는데, 특히 수학이나 instruction 평가에서 파싱 오류를 방지하기 위해 필수적입니다.
```
Comment (ahnjj) [inline _posts/2025-11-3-Welcome-GPT-OSS.md:357]:
```suggestion
GPT OSS 모델은 추론 모델입니다. 따라서 평가 할 때 필요한 생성 크기(최대 새 토큰 수)가 매우 큽니다. 모델의 생성 결과에는 먼저 추론 과정이 포함되고, 그 다음에 실제 답변이 나오기 때문입니다. 생성 크기가 너무 작으면 추론 중간에 예측이 중단될 위험이 있으며, 이는 위음성을 발생시킬 수 있습니다. 메트릭을 계산하기 전에 모델 답변에서 추론 과정을 제거해야 하는데, 특히 수학이나 instruction 평가에서 파싱 오류를 방지하기 위해 필수적입니다.
```
자연스럽게 앞뒤를 바꾸어 보았어요~!

**Review by ahnjj — APPROVED** [review]
흥미로운 주제 알려주셔서 감사해요~! 아주 사소한 부분만 리뷰드립니다 ㅎㅎ

File: _posts/2025-11-3-Welcome-GPT-OSS.md (hunk 3 above)
Comment (choincnp) [reply _posts/2025-11-3-Welcome-GPT-OSS.md]: 감사합니다 :) 제가 누락된 부분이 있었네용 ㅠ

File: _posts/2025-11-3-Welcome-GPT-OSS.md (hunk 4 above)
Comment (choincnp) [reply _posts/2025-11-3-Welcome-GPT-OSS.md:26]: 안녕하세요 효정님, 뒤에 Acknowledgements는 일부러 누락시켰습니다! 오탈자 및 잘못 번역한것들 검수해주셔서  감사합니다 :)

File: _posts/2025-11-3-Welcome-GPT-OSS.md (hunk 9 above)
Comment (choincnp) [reply _posts/2025-11-3-Welcome-GPT-OSS.md]: 안녕하세요 정님, 꼼꼼한 번역검수 감사드립니다 :) !!
